tidy3d/components/grid/grid.py

```diff
@@ -350,7 +350,7 @@ def discretize_inds(self, box: Box, extend: bool = False) -> List[Tuple[int, int
             inds_leq_pt_min = np.where(bound_coords <= pt_min)[0]
             ind_min = 0 if len(inds_leq_pt_min) == 0 else inds_leq_pt_min[-1]
 
-            if extend:
+            if extend and ind_max > ind_min:
```

Comment:
Not sure I understood what this does: is the idea that `ind_max` and `ind_min` will be equal if the monitor has a zero dim, and in that case the grid is not extended even if `extend` is `True`, because interpolation will not be needed?

Reply:
Actually no, this is a fix to the case when the geometry is entirely outside of the simulation domain. The extend part would error in that case, instead now we return the indexes which are such that `ind_min:ind_max` is empty.